File: mindflow/src/command_helpers/diff.rs
use std::process::Command;
use crate::utils::prompts::GIT_DIFF_PROMPT_PREFIX;

// Generate a git diff prompt by executing the git diff command and appending the result to the GIT_DIFF_PROMPT_PREFIX
pub async fn generate_diff_prompt(diffargs: &Vec<String>) -> String {
    let command = ["git", "diff"]
        .iter()
        .map(|s| s.to_string())
        .chain(diffargs.iter().map(|s| s.to_string()))
        .collect::<Vec<_>>();

    let diff_result = Command::new(&command[0])
        .args(&command[1..])
        .output()
        .expect("failed to execute process");
    
    format!("{}\n\n{}", GIT_DIFF_PROMPT_PREFIX, String::from_utf8_lossy(&diff_result.stdout))
}

File: mindflow/command_helpers/diff/diff.py
import subprocess
from typing import List

from mindflow.utils.prompts import GIT_DIFF_PROMPT_PREFIX


def generate_diff_prompt(diffargs: List[str]) -> str:
    """
    This function is used to generate a prompt based on a diff.
    """
    command = ["git", "diff"] + diffargs

    # Execute the git diff command and retrieve the output as a string
    diff_result = subprocess.check_output(command).decode("utf-8")

    return f"{GIT_DIFF_PROMPT_PREFIX}\n\n{diff_result}"
